server/asset/eth/coiner.go

```diff
@@ -147,23 +154,13 @@ func (backend *Backend) newSwapCoin(coinID []byte, contractData []byte, sct swap
 		return nil, fmt.Errorf("unable to convert value: %v", err)
 	}
 
-	// For redemptions, the transaction should move no value.
-	if sct == sctRedeem && value != 0 {
-		return nil, fmt.Errorf("expected swapCoin value of zero for redeem but got: %d", value)
-	}
```

Comment:
This check removed. Not needed? Any redeem tx with a value should fail to redeem because the solidity `redeem` function is not marked payable.

Reply:
Value is now being checked (to a tighter standard) in `newSwapCoin`.

Reply:
Isn't that for inits, I mean redeems should be 0 value.